docs/design-docs/04-arm-build.md

```diff
@@ -0,0 +1,66 @@
+# Title
+
+* Author(s): Mike Borozdin, @mikebz
+* Approver: Mengqi, Sunil
+
+## Why
+
+More people are getting Apple M1 machines.  The current docker images for
+functions do not work.  Here is an example:
+https://github.com/GoogleContainerTools/kpt/issues/2874
+While it's not a problem for CI/CD pipelines where architecture is mostly amd64
+for client development purposes the users are stuck.
+
+## Design
+
+Building arm64 and amd64 multi platform images is possible:
+https://www.docker.com/blog/multi-arch-build-and-images-the-simple-way/
+but it has it's caveats.  The build is done in a builder which needs to be 
+created:
+https://docs.docker.com/engine/reference/commandline/buildx_create/
+The caveat is that the multi platform images do not load into docker images:
+https://github.com/docker/buildx/blob/master/docs/reference/buildx_build.md#-load-the-single-platform-build-result-to-docker-images---load
+
+Right now the build pipeline has several steps:
+1) build
+2) tag
+3) push
+
+In different scenarios only the first step is executed.  If we switch to 
```

Comment:
`build` and `tag` are usually done together. No matter if the users are going to push the images, they almost always tags images.